### DISCUSSION TITLE
Selectors with classes not working in layers

### DISCUSSION THREAD

**Discussion by luchillo17** [tailwindlabs/tailwindcss#13723] [Help] [answered]
Here's the issue, see the `MuiDataGrid-menu` in the next code block? if I remove it and only have the `body > ...` selector, it is not included in the CSS, it's not even a real tag in use anywhere, could have used anything like `randomTag6969` and would have worked the same, and it has the same issue regardless of layer (tried in `@layer base` to be safe, still get the issue):
```css
@layer components {
  /* #region MUI overrides */
  MuiDataGrid-menu,
  body > div.base-Popper-root.MuiDataGrid-menu {
    @apply z-50;
  }
  /* #endregion */
}
```

This also breaks, not sure why but if I use just classes selectors, the same issue happens:
```css
@layer components {
  /* #region MUI overrides */
  .base-Popper-root.MuiDataGrid-menu {
    @apply z-50;
  }
  /* #endregion */
}

**Comment by crswll** [comment]
If you put it in a `@layer` Tailwind CSS will make sure it's used before adding it to your CSS. If you want it to show up even if the classes don't exist in your markup you can just not put it in a layer or add the MUI code paths to your Tailwind CSS config `content` section.

PS: You're missing a `.` before `MuiDataGrid-menu` in the first bit of code which would explain why it still shows up for that first example.

**Reply by luchillo17** [reply]
HI @crswll, thanks for a quick answer, I guess I would need to add the MUI `node_modules` path for this specific MUI component?.

Also, I did not forget the `.`, it's intentional, if I do a `.MuiDataGrid-menu`, Tailwind will drop it since it's not in use in my code, it's in MUI so my code never sees that class ever.

**Reply by crswll** [reply, marked as answer]
Yes. Check out: https://tailwindcss.com/docs/content-configuration#working-with-third-party-libraries

**Reply by luchillo17** [reply]
I see, thanks a lot, I'll end up removing this anyway because of a different issue with mobile tap events bubbling up to the MUI dropdown which closes my dialog, but still, good to know how it's done properly.
